### PR TITLE
metrics: change resource group label name

### LINKED ISSUE #16191: unify the metrics label of resource group name
## Development Task

In order to add a new global label selector for resource-group-name, we should first unify the label name of resource-group-name in all places and change it to a unique one.

Currently, most of the label name is "name" which is so commonly used, we will change it to `resource_group` in all repos.

### PR DESCRIPTION
**Pull request by glorv**
### What is changed and how it works?

Issue Number: Close #16191

What's Changed:

```commit-message
change metrics label name for resource-group-name from "name" to "resource_group".

To be backward compatible with old grafana panel, we add a new label name and keep the old one. We are going to deprecate the old label in v8.0.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/resource_control/src/metrics.rs [hunk 1]
```diff
@@ -4,22 +4,25 @@ use lazy_static::*;
 use prometheus::*;
 
 lazy_static! {
+    // TODO: deprecate the "name" label in v8.0.
```
Comment (nolouch) [inline components/resource_control/src/metrics.rs:7]: I see some metrics not introduced in the Grafana template. maybe we can directly rename some metrics in `TiKV`

**Comment by bufferflies** [comment]
the grafana.json file needs to change also.

File: components/resource_control/src/metrics.rs (hunk 1 above)
Comment (glorv) [reply components/resource_control/src/metrics.rs:7]: All the metrics changed in this PR are used in grafana. The background related metrics panels are defined in the Resource-Control page in tidb repo.

**Comment by glorv** [comment]
> the grafana.json file needs to change also.

@bufferflies We want to keep backward compatibility with v7.x so the expression in grafana is not changed. Maybe we can change it in v8.0

**Review by nolouch — APPROVED** [review]

File: components/resource_control/src/metrics.rs (hunk 1 above)
Comment (Connor1996) [inline components/resource_control/src/metrics.rs:7]: why not replace it directly, grafana panel should be updated with the version

Comment (glorv) [reply components/resource_control/src/metrics.rs:7]: There is a problem that if we replace it, after a cluster is upgraded from old version, old metrics data will lost due to new expression. @nolouch  do you think we should keep this compatibility?

**Review by Connor1996 — APPROVED** [review]
LGTM

File: components/resource_control/src/metrics.rs (hunk 1 above)
Comment (nolouch) [reply components/resource_control/src/metrics.rs:7]: IMO, the background relatives can be replaced directly due to no cases using it currently. if we do we need to pick to release-7.5

**Pull request merged**